internal/controller/stas/containerimagescan_controller.go

```diff
@@ -84,6 +100,36 @@ func (r *ContainerImageScanReconciler) backoffDuration(lastScan *metav1.Time, no
 	return time.Minute + time.Duration(float64(time.Minute)*priority)
 }
 
+// latestDigestScan returns the most recently scanned CIS with the specified digest.
+func (r *ContainerImageScanReconciler) latestDigestScan(ctx context.Context, dig digest.Digest) (*stasv1alpha1.ContainerImageScan, error) {
+	cisList := &stasv1alpha1.ContainerImageScanList{}
+
+	listOps := []client.ListOption{
+		client.MatchingFields{indexDigest: string(dig)},
+	}
+	if err := r.List(ctx, cisList, listOps...); err != nil {
+		return nil, err
+	}
+
+	var (
+		latest   *stasv1alpha1.ContainerImageScan
+		scanTime *time.Time
+	)
+
+	for _, c := range cisList.Items {
+		if c.Status.LastSuccessfulScanTime == nil {
+			continue
+		}
+
+		if scanTime == nil || c.Status.LastSuccessfulScanTime.After(*scanTime) {
+			scanTime = &c.Status.LastSuccessfulScanTime.Time
+			latest = &c
```

Comment:
```suggestion
			latest = &c
        break
```

Reply:
We could, as the currently found is valid, but thought it would be nice to take the latest one if more valid ones exist

Reply:
No wait, the validity check (more recent than scan interval) is not here, this fn just finds the latest one. So if breaking, we would also need to check that it's less than recanInterval old here.

Reply:
Although likely not an issue, sorting is a lot of extra work to just to find the lastest one.

You have a point in that when multiple sufficiently recent ones are found, they are likely at the same timestamp or not far apart (as multiple jobs are scheduled at the same time, the same digest can still be scanned multiple times).